src/api/schema/events/log.rs

```diff
@@ -0,0 +1,31 @@
+use crate::event;
+use async_graphql::Object;
+use serde_json::json;
+
+#[derive(Debug)]
+/// A log event
+pub struct LogEvent(event::LogEvent);
+
+impl LogEvent {
+    pub fn new(ev: event::LogEvent) -> Self {
+        Self(ev)
+    }
+}
+
+#[Object]
+impl LogEvent {
+    /// Log message
+    async fn message(&self) -> Option<String> {
+        serde_json::from_value(json!(self.0.get("message")?)).ok()
```

Comment:
This seems a bit round the houses to extract a String. Is there a way to avoid the initial serialisation to a JSON string? I'm pretty sure I'm missing something obvious.

Reply:
I think you probably want to match:

```rust
match self.0.get("message").and_then(|message| match {
  v @ Value::Bytes(_) => Some(v.to_string_lossy()),
  _ => None,
})
```

Reply:
Thanks! I actually missed an existing method of `event::Value` that reduces this to:

```rust
Some(self.0.get("message")?.to_string_lossy())
```